src/Backends/IF97/IF97Backend.h

```diff
@@ -31,24 +31,62 @@ class IF97Backend : public AbstractState  {
     void set_volu_fractions(const std::vector<CoolPropDbl> &volu_fractions){throw NotImplementedError("Volume composition has not been implemented.");};
     const std::vector<CoolPropDbl> & get_mole_fractions(void){throw NotImplementedError("get_mole_fractions composition has not been implemented.");};
 
-    /// Updating function for incompressible fluid
+    /// Override clear() function of IF97 Water
+    bool clear() {
+    // Reset all instances of CachedElement and overwrite
+    // the internal double values with -_HUGE
+        this->_rhomolar = -_HUGE;
+        this->_T = -_HUGE;
+        this->_p = -_HUGE;
+        this->_Q = -_HUGE;
+        this->_phase = iphase_not_imposed; // Default condition is no phse imposed
+
+        return true;
+    };
+
+    /// Updating function for IF97 Water
     /**
     In this function we take a pair of thermodynamic states, those defined in the input_pairs
-    enumeration and update all the internal variables that we can.
+    enumeration and update the bare minimum values needed to calculate the other values.
 
     @param input_pair Integer key from CoolProp::input_pairs to the two inputs that will be passed to the function
     @param value1 First input value
     @param value2 Second input value
     */
     void update(CoolProp::input_pairs input_pair, double value1, double value2){
+        
+        clear();  //clear the few cached values we are using
+
         switch(input_pair){
             case PT_INPUTS: _p = value1; _T = value2; break;
-            /* TODO:  Add input pairs for saturation functions 
-            case PQ_INPUTS: _p = value1; _Q = value2; break;
-            case QT_INPUTS: _Q = value1; _T = value2; break;
-            */
+            case PQ_INPUTS: 
+                _p = value1; 
+                _Q = value2; 
+                _T = IF97::Tsat97(_p); 
+                if (std::abs(_Q) < 1e-10){
+                    _phase = iphase_liquid; // bubble point
+                }
+                else if (std::abs(value2-1) < 1e-10){
+                    _phase = iphase_gas; // dew point
+                }
+                else
+                    _phase = iphase_twophase;
+                break;
+            case QT_INPUTS: 
+                _Q = value1; 
+                _T = value2; 
+                _p = IF97::psat97(_T); 
+                if (std::abs(_Q) < 1e-10){
+                    _phase = iphase_liquid; // bubble point
```

Comment:
Technically yes, but I'm trying to make only one saturated function call and not two in the calc_Flash function and need the flag.  If at the bubble or dew point, I only want to call satL() or satV(), not both to minimize the real computations.  Not a big deal normally unless you're calculating a million points; then it saves 50% of the computational time.  This contributes to your third comment.  I could set, the phase to iphase_two_phase and do the checks for Q ==0 or Q == 1 in the calc_Flash routine.  It just looks cleaner this way AND its for a PQ/QT input pair so we should only be in the vapor dome anyway.

I can move the logic if you think the semantics will create confusion.  IF97 doesn't use these flags internally so it won't care.  LMK